src/session/session.c

```diff
@@ -948,6 +944,8 @@ main (int argc,
 
   /* And get back the authorization header */
   char *authorization = read_authorize_response ("authorization");
+  char *rhost = get_authorize_key (authorization, "remote-peer", FALSE);
```

Comment:
Okay.  Here's the part I was actually intending to talk about when I confusingly mentioned about the "required" key thing above.  With `required` set to `FALSE` here this could return `NULL`, which is unsafe because `perform_basic()` assumes that this must be non-NULL.  The `getenv()` version of this code used `""` as a default value.

Reply:
Right, thanks! Fixed. It didn't crash anywhere in tests because cockpit-ws now unconditionally passes it (with a similar `""` fallback), but let's be robust.

Reply:
A bit more detail on this claim: the `rhost` field is filled in when creating the `CockpitCreds` struct.  That happens with `cockpit_creds_new()`.  That gets called from exactly two places.  In one place (`cockpit_auth_local_async()`) it's hardcoded to `"localhost"`.  In the other (the "real" case) it comes from `cockpit_web_request_get_remote_address()`.

But: this function can definitely return `NULL`.  In fact, it has serveral branches that literally `return NULL;`.  Are those dead code?

I think with the `""` fallback we're on the right side of "safe" here, but I think we might benefit from pulling on this particular thread.  Maybe we find some dead code we can delete, or some `if` that we can turn into `assert`.

Reply:
> `perform_basic()` assumes that this must be non-NULL. 

[pam_set_item()](https://linux.die.net/man/3/pam_set_item) is documented to be fine with a NULL rhost. But `btmp_log()` indeed does an unchecked `strncpy()` on it. Similarly, session.c calls `utmp_log()` which in turn calls `do_lastlog()`, both of which  also assume a non-NULL rhost. AFAICS cockpit-session is fine as-is, but let's add some assertions to clarify/document this assumption.

On the ws side, CockpitCreds is completely ignorant about a NULL rhost, setting `COCKPIT_CRED_RHOST, NULL` has the same effect as not setting the property at all. So the two callers are fine.  `cockpit_session_launch()` already doesn't set `$COCKPIT_REMOTE_PEER` if it's unknown -- it's just that this never actually seems to be the case in our tests, it always knows the peer. So we *could* tighten this assumption into an assert, but TBH it feels right to be defensive there. So the only thing to fix there IMHO is doc/authentication.md.

I didn't find anything else to clean up here.

These two changes are too small for their own PR, so I'll just add them here.